test-render

```diff
@@ -74,6 +74,16 @@ somerandomtextmorerandomtext
 \$symbol_that_doesnt_get_rendered
 EOF
 
+it "does throw an error if test_variable is defined but empty if --no-empty flag is provided" "test_variable=" "--no-empty" <<EOF
+ERROR: test_variable is empty
+EOF
+
+it "doesn' throw an error if test_variable is defined but empty and --no-empty flag is provided on last position" "test_variable=" "" "--no-empty" <<EOF
```

Comment:
Thanks for adding the tests, @Globegitter! I was going to argue that it should work no matter where the flag is placed but thankfully I had the sense to review my POSIX and realized that you were right in doing this. :-)

Just one tiny nitpick though: Can you add the `t` to `doesn'` above?